activejob/lib/active_job/queue_adapters/async_adapter.rb

```diff
@@ -0,0 +1,38 @@
+require 'active_job/async_job'
+
+module ActiveJob
+  module QueueAdapters
+    # == Async adapter for Active Job
+    #
+    # When enqueueing jobs with Async Job each job will be executed
+    # asynchronously on a +concurrent-ruby+ thread pool. All job data
+    # is retained in memory. Because job data is not saved to a persistent
+    # datastore there is no additional infrastructure needed and most jobs
+    # will process very quickly. The lack of persistence, however, means
+    # that all unprocessed jobs will be lost on application restart. This
+    # makes in-memory queue adapters unsuitable for most production
+    # environments but excellent for development and testing.
+    #
+    # Concurrent Ruby: Modern concurrency tools including agents, futures,
+    # promises, thread pools, supervisors, and more. Inspired by Erlang,
+    # Clojure, Scala, Go, Java, JavaScript, and classic concurrency patterns.
```

Comment:
You can certainly simply specify that this adapter creates `AsyncJob` objects under the hood.